server/server.go

```diff
@@ -365,16 +396,25 @@ func (server *Server) checkNetIsolation() {
 // handleSessionClose handles the closing of a session. It removes the player of the session from the server.
 func (server *Server) handleSessionClose(controllable session.Controllable) {
 	server.playerMutex.Lock()
+	if p, ok := server.p[controllable.UUID()]; ok {
+		err := server.playerProvider.Save(controllable.UUID(), p.Data())
+		if err != nil {
+			server.log.Errorf("Error while saving data: %v", err)
+		}
```

Comment:
Would consider moving this outside of the mutex lock so you don't have to keep that locked while saving data.